File: Service/Translators/UnitTypeTranslator.vb
Imports BO
Imports DAL

Public Class UnitTypeTranslator
    Public Shared Function TranslateEntityToBO(_entity As UnitTypes, bo As UnitTypeBO) As ErrorCode
        If _entity Is Nothing Then Return ErrorCode.EntityNull
        If bo Is Nothing Then Return ErrorCode.BoNull
        bo.Id = _entity.ID
        bo.Name = _entity.Name
        bo.Shortcode = _entity.Shortcode
        bo.GroupId = _entity.GroupID
        Return ErrorCode.Success
    End Function

    Friend Shared Function TranslateBOToEntity(_entity As UnitTypes, bo As UnitTypeBO) As ErrorCode
        If _entity Is Nothing Then Return ErrorCode.EntityNull
        If bo Is Nothing Then Return ErrorCode.BoNull
        _entity.Name = bo.Name
        _entity.Shortcode = bo.Shortcode
        If (bo.GroupId <> 0) Then
            _entity.GroupID = bo.GroupId
        End If
        Return ErrorCode.Success
    End Function
End Class

File: ServiceCore/Translators/UnitTypeTranslator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using BOCore;
using DALCore;
using DALCore.Models;

namespace ServiceCore.Translators
{
    public class UnitTypeTranslator
    {
        public static ErrorCode TranslateEntityToBO(UnitTypes _entity, UnitTypeBO bo)
        {
            if (_entity == null)
                return ErrorCode.EntityNull;
            if (bo == null)
                return ErrorCode.BoNull;
            bo.Id = _entity.Id;
            bo.Name = _entity.Name;
            bo.Shortcode = _entity.Shortcode;
            bo.GroupId = _entity.GroupId;
            return ErrorCode.Success;
        }

        internal static ErrorCode TranslateBOToEntity(UnitTypes _entity, UnitTypeBO bo)
        {
            if (_entity == null)
                return ErrorCode.EntityNull;
            if (bo == null)
                return ErrorCode.BoNull;
            _entity.Name = bo.Name;
            _entity.Shortcode = bo.Shortcode;
            if ((bo.GroupId != 0))
                _entity.GroupId = bo.GroupId;
            return ErrorCode.Success;
        }
    }
}
